presto-tpcds/src/main/java/com/facebook/presto/tpcds/TpcdsRecordSet.java

```diff
@@ -0,0 +1,188 @@
+import com.facebook.presto.spi.type.DecimalType;
+import com.facebook.presto.spi.type.Decimals;
+import com.facebook.presto.spi.type.Type;
+import com.google.common.collect.ImmutableList;
+import com.teradata.tpcds.Results;
+import com.teradata.tpcds.column.Column;
+import com.teradata.tpcds.column.ColumnType;
+import io.airlift.slice.Slice;
+import io.airlift.slice.Slices;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Iterator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.spi.type.Decimals.rescale;
+import static com.facebook.presto.tpcds.TpcdsMetadata.getPrestoType;
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.Double.parseDouble;
+import static java.lang.Integer.parseInt;
+import static java.lang.Long.parseLong;
+import static java.util.Objects.requireNonNull;
+
+public class TpcdsRecordSet
+        implements RecordSet
+{
+    private final Results results;
+    private final List<Column> columns;
+
+    private final List<Type> columnTypes;
+
+    public TpcdsRecordSet(Results results, List<Column> columns)
+    {
+        requireNonNull(results, "results is null");
+
+        this.results = results;
+        this.columns = ImmutableList.copyOf(columns);
+        this.columnTypes = ImmutableList.copyOf(this.columns.stream().map(column -> getPrestoType(column.getType())).collect(Collectors.toList()));
+    }
+
+    @Override
+    public List<Type> getColumnTypes()
+    {
+        return columnTypes;
+    }
+
+    @Override
+    public RecordCursor cursor()
+    {
+        return new TpcdsRecordCursor(results.iterator(), columns);
+    }
+
+    public class TpcdsRecordCursor
+            implements RecordCursor
+    {
+        private final Iterator<List<List<String>>> rows;
+        private final List<Column> columns;
+        private List<String> row;
+        private boolean closed;
+
+        public TpcdsRecordCursor(Iterator<List<List<String>>> rows, List<Column> columns)
+        {
+            this.rows = rows;
+            this.columns = columns;
+        }
+
+        @Override
+        public long getTotalBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getCompletedBytes()
+        {
+            return 0;
+        }
+
+        @Override
+        public long getReadTimeNanos()
+        {
+            return 0;
+        }
+
+        @Override
+        public Type getType(int field)
+        {
+            return getPrestoType(columns.get(field).getType());
+        }
+
+        @Override
+        public boolean advanceNextPosition()
+        {
+            if (closed || !rows.hasNext()) {
+                closed = true;
+                row = null;
+                return false;
+            }
+
+            row = rows.next().get(0);
+            return true;
+        }
+
+        @Override
+        public boolean getBoolean(int field)
```

Comment:
yeah if the java type for your column was boolean this would get the value from the results and return the appropriate boolean.  TPC-DS doesn't have any boolean columns, though